src/scene/graph-node.js

```diff
@@ -1103,11 +1116,29 @@ pc.extend(pc, function () {
                 node._notifyHierarchyStateChanged(node, enabledInHierarchy);
             }
 
+            // The graph depth of the child and all of its descendants will now change
+            node._updateGraphDepth();
```

Comment:
This is probably OK. But we should watch out for how many times this is called during initialization. We may want to trigger it once at the end of setup instead of everything a entity is added.

Reply:
I did some testing – it seems that during initialization it gets called ~3 times for each node when running in the editor, and ~4 times when running in a game (i.e. via Launch). For comparison, `_dirtify()` is called roughly the same number of times.

Do you think it's worth changing it so that it's recalculated on demand?

Reply:
I think it's fine